src/utils/retry-options.ts

```diff
@@ -0,0 +1,24 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {BackoffSettings} from 'google-gax/build/src/gax';
+
+// See protos/google/rpc/code.proto
+// (4=DEADLINE_EXCEEDED, 8=RESOURCE_EXHAUSTED, 10=ABORTED, 14=UNAVAILABLE)
```

Comment:
First bullet point:
I made this change to preserve [previous client library behaviour](https://github.com/googleapis/nodejs-bigtable/blob/724b7118972e6697e3fd648b5d6ce2d047ab2b6a/src/table.ts#L49C7-L49C29). I looked at git history and it looks like these retry codes were added during a customer escalation so I think asking CBT engineers would be a good idea.

Second bullet point:
Thanks for the reminder. I think this is directly related to a customer requirement. I see the generated config is `non_idempotent` which is no retry codes and that doesn't seem ideal though I could be wrong. That depends on customer expectations.

Reply:
This differs a bit from the [retryable errors I was told](https://github.com/googleapis/python-bigtable/blob/2a2bbfdba6737c508ab1073d37fef680ca2a8c2f/google/cloud/bigtable/data/_async/client.py#L449) when building the Python version (we retry deadline_exceeded and Unavailable on all rpcs, and aborted only on read_rows. And then you need to make sure you're only retrying idempotent operations)

It might be worth conforming with the Bigtable team to make sure this is still the right configuration

Reply:
I asked about this in YAQs